app/src/main/java/com/example/caregiver/Task_Caregivee.java

```diff
@@ -0,0 +1,195 @@
+    protected void loadCaregiveesTask() {
+        SharedPreferences preferences = PreferenceManager.getDefaultSharedPreferences(getActivity());
+        String caregiveeId = preferences.getString("userId", "");
+        if (caregiveeId.isEmpty()) {
+            return;
+        }
+        DatabaseReference database = FirebaseDatabase.getInstance().getReference();
+        DatabaseReference ref = database.child("users/" + caregiveeId);
+        ref.addValueEventListener(new ValueEventListener() {
+            @RequiresApi(api = Build.VERSION_CODES.N)@Override
+        public void onDataChange(@NonNull DataSnapshot snapshot) {
+            Object roomObject = snapshot.child("rooms").getValue();
+            if (roomObject != null) {
+                Gson gson = new Gson();
+                String tasksJson = gson.toJson(roomObject);
+                taskList = createTaskList(tasksJson);
+                displayTaskList(taskList);
+            }
+        }@Override
+        public void onCancelled(@NonNull DatabaseError error) {
+            Log.d("error", "Can't query caregivees for this caregiver");
+        }
+        });
+    }
+
+    /**
+     * Returns all tasks associated with that caregivee.
+     * @param roomString Json-string representation of all the rooms.
+     */
+    @RequiresApi(api = Build.VERSION_CODES.N)
+    protected List < Task > createTaskList(String roomString) {
+
+        // Initialize an array list that will store all tasks associated with the caregivee.
+        List < Task > tasks = new ArrayList < >();
+
+        // Parse the roomString to return a json Object representation.
+        JsonParser parser = new JsonParser();
+        JsonObject roomObject = (JsonObject) parser.parse(roomString);
+        List < String > rooms = roomObject.entrySet().stream().map(
+                i ->i.getKey()).collect(Collectors.toCollection(ArrayList::new));
+
+        // For each room, get their corresponding tasks
+        for (String roomStr: rooms) {
+            JsonObject singleRoom = roomObject.getAsJsonObject(roomStr);
+            JsonObject tasksPerRoom = singleRoom.getAsJsonObject("tasks");
+            if (tasksPerRoom != null) {
+                List < String > tasksIds = tasksPerRoom.entrySet().stream().map(
+                        i ->i.getKey()).collect(Collectors.toCollection(ArrayList::new));
+
+                // For each task, put them in the Task object.
+                for (String taskId: tasksIds) {
+                    Log.d("taskId", taskId);
+                    JsonObject task = tasksPerRoom.getAsJsonObject(taskId);
+                    String caregiverId = task.get("caregiverID").toString();
+                    String taskName = task.get("name").toString();
+                    String taskNote = task.get("notes").toString();
+                    String assignedStatus = task.get("assignedStatus").toString();
+                    String completionStatus = task.get("completionStatus").toString();
```

Comment:
I believe this way of retrieving the String has it in quotes twice like this -> ""name"". I found this method `task.get("name").getAsString();` to fix that in TaskFragment

Reply:
oh that's pretty cool, I will fix that.